### PR TITLE
Lau 933 access request get request

### PR DESCRIPTION
**Pull request by rapolaskaseliscgi**
### JIRA link (if applicable) ###
LAU-933

### Change description ###
Added GET access request endpoint

### CODE REVIEW

**Review by sonny-hmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/laubackend/cases/response/AccessRequestGetResponse.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+@Builder
+public class AccessRequestGetResponse {
+
+    private List<AccessRequestLog> accessLog;
+
+    private int startRecordNumber;
+
+    private boolean moreRecords;
+
+    private long totalNumberOfRecords;
```
Comment (priyankaVerma21) [inline src/main/java/uk/gov/hmcts/reform/laubackend/cases/response/AccessRequestGetResponse.java:21]: Total number of records was not there in LLD , If returning we need to include in LLD as we are doing for other end points

File: src/test/java/uk/gov/hmcts/reform/laubackend/cases/controllers/AccessRequestControllerTest.java [hunk 2]
```diff
@@ -75,4 +80,56 @@ void shouldReturnInternalServerErrorForBadPostRequest() {
...
+            .endTimestamp("2024-06-24T22:20:05.000")
+            .build();
+        ResponseEntity<AccessRequestGetResponse> responseEntity = accessRequestController.getAccessRequest(
+            null,
+            null,
+            request
+        );
+
+        verify(accessRequestService, times(1)).getAccessRequestRecords(request);
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
```
Comment (priyankaVerma21) [inline src/test/java/uk/gov/hmcts/reform/laubackend/cases/controllers/AccessRequestControllerTest.java:103]: This test should fail as userId or requestType is not provided which is mandatory

Comment (rapolaskaseliscgi) [reply src/test/java/uk/gov/hmcts/reform/laubackend/cases/controllers/AccessRequestControllerTest.java:103]: In my understanding, one or more must be present of those three, meaning if caseRef is there (and both timestamps) it should be fine.

File: src/main/java/uk/gov/hmcts/reform/laubackend/cases/response/AccessRequestGetResponse.java (hunk 1 above)
Comment (rapolaskaseliscgi) [reply src/main/java/uk/gov/hmcts/reform/laubackend/cases/response/AccessRequestGetResponse.java:21]: Yes, it wasn't there, but I decided to add it in order to be consistent with the other GET endpoints.

File: src/test/java/uk/gov/hmcts/reform/laubackend/cases/controllers/AccessRequestControllerTest.java (hunk 2 above)
Comment (priyankaVerma21) [reply src/test/java/uk/gov/hmcts/reform/laubackend/cases/controllers/AccessRequestControllerTest.java:103]: That's fine

**Review by priyankaVerma21 — APPROVED** [review]

**Pull request merged**